### ISSUE TITLE
Ability to set global overrides for `cyclopts.Parameters`

### ISSUE THREAD

**Issue by Ravencentric** [BrianPugh/cyclopts#24]
Take this case as an example:
I do not want negatives or defaults to show up in --help for everything but one. Currently, I gotta do something like:
```py
foo: Annotated[bool, Parameter(name="--foo", help="do foo", negative="", show_default=False)] = False,
bar: Annotated[bool, Parameter(name="--bar", help="do bar", negative="", show_default=False)] = False,
baz: Annotated[bool, Parameter(name="--baz", help="do baz")] = False,
```
Is it (or would it be) possible to set some global default overrides? 
Example:

```py
cyclopts.GlobalParameters(negative="", show_default=False)

foo: Annotated[bool, Parameter(name="--foo", help="do foo")] = False,
bar: Annotated[bool, Parameter(name="--bar", help="do bar")] = False,
# override show_default global for baz but negative continues to follow global default
baz: Annotated[bool, Parameter(name="--baz", help="do baz", show_default=True)] = False,
```

**Comment by BrianPugh** [comment]
I'm actually already working on this feature! I think it's going to look like this:

```
from cyclopts import App, Parameter

app = App(default_parameter=Parameter(negative="", show_default=False))

# Now all Parameters have `negative=""` and `show_default=False` as the default.
```

Any feedback on this proposed solution?

**Comment by Ravencentric** [comment]
Looks perfect, thank you!

**Comment by BrianPugh** [comment]
This might have to wait a week or two. There's a [newly merged feature in attrs](https://github.com/python-attrs/attrs/pull/1187) that significantly simplifies the implementation. [Hynek intends to do a release before new years](https://github.com/python-attrs/attrs/issues/1212#issuecomment-1858940052).

**Comment by BrianPugh** [comment]
I found a workaround via the [`record_init_kwargs` decorator](https://github.com/BrianPugh/cyclopts/pull/33/files#diff-843c8548fd7abe5479a26f75ba52b1c4443c3014d18821dbdcd6e7f830bc5fbeR4).

**Comment by BrianPugh** [comment]
This has been and released in v1.2.0.
